server/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -471,7 +471,7 @@ public NodeIndicesStats stats(CommonStatsFlags flags) {
             }
         }
 
-        return new NodeIndicesStats(commonStats, statsByIndex(this, flags), statsByShard(this, flags));
+        return new NodeIndicesStats(commonStats, statsByIndex(this, flags), needShardsStats ? statsByShard(this, flags) : new HashMap<>(0));
```

Comment:
This isn't going to work I think, we still have to run `statsByShard()` to collect the shard-level stats or else the totals will all be zero. I will kick off a CI run to show you some failing tests.

Reply:
You are right.`statsByShard ` is also used in `NodeIndicesStats()`. Fixed in e3e2624372027989d67798a9fdedb71ecc41f880